C-Plus-Plus/cp/Knuth–Morris–Pratt.cpp

```diff
@@ -0,0 +1,110 @@
+// https://en.wikipedia.org/wiki/Knuth–Morris–Pratt_algorithm
+
+// Time Complexity:-O(N+M).
+// Space Complexity:-O(M).
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+using namespace std;
+
+void computeLPS(string &, vector<int> &);
+void searchByKMP(string &, string &);
+
+int main()
+{
+    string text = "", patt = "";
+
+    int n;
+
+    cout << "Enter Number of test Cases:";
+    cin >> n;
+
+    while (n)
+    {
+        cout << endl;
+        cout << "Enter the text string:";
+        cin >> text;
+
+        cout << endl
+             << "Enter the pattern string:";
```

Comment:
@anshul7sh format this line like this `cout << endl << "Enter the pattern string:";`

Reply:
@iamrajiv Code Updated as mentioned.